### PR TITLE
Mod quiz

### PR DESCRIPTION
**Pull request by milt**
mod_quiz statement fixes and additions.

Note that question_attempt_restarted was not firable and has not been included.

### CODE REVIEW

File: src/transformer/utils/result/get_matching_response.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ */
+function get_matching_response(
+    array $config,
+    \stdClass $questionattempt,
+    string $lang
+) {
+    $selmap = array_reduce(
+        explode('; ', $questionattempt->responsesummary),
+        function ($reduction, $selection) {
+            $split = explode("\n -> ", $selection);
```
Comment (cliffcaseyyet) [inline src/transformer/utils/result/get_matching_response.php:44]: what the what? What did this look like to start with?

File: src/transformer/get_event_function_map.php [hunk 2]
```diff
@@ -119,7 +119,7 @@ function get_event_function_map() {
         '\mod_lesson\event\lesson_resumed' => 'mod_lesson\lesson_resumed',
         '\mod_lesson\event\lesson_restarted' => 'mod_lesson\lesson_restarted',
         '\mod_lesson\event\lesson_ended' => 'mod_lesson\lesson_ended',
-        '\mod_lesson\event\question_viewed' => 'mod_lesson\question_viewed',
```
Comment (cliffcaseyyet) [inline src/transformer/get_event_function_map.php:122]: my bad

Comment (milt) [reply src/transformer/get_event_function_map.php:122]: wait, what even happened here

File: src/transformer/utils/result/get_matching_response.php (hunk 1 above)
Comment (milt) [reply src/transformer/utils/result/get_matching_response.php:44]: this code is barely touched, just moved it

File: src/transformer/get_event_function_map.php (hunk 2 above)
Comment (milt) [reply src/transformer/get_event_function_map.php:122]: ohhh spaces yeah

**Review by cliffcaseyyet — APPROVED** [review]

**Pull request merged**